# Activity_Week06

#### Dataset used : Dataset_w6
#### 196 rows × 7 columns

In [1]:
import numpy as np
import pandas as pd

In [2]:
data=pd.read_excel('Dataset_w6.xlsx')
data

country             Region  \
0                           Afghanistan       SouthernAsia   
1                               Albania     SouthernEurope   
2                               Algeria     NorthernAfrica   
3                                Angola       MiddleAfrica   
4                   Antigua and Barbuda          Caribbean   
..                                  ...                ...   
191  Venezuela (Bolivarian Republic of)       SouthAmerica   
192                            Viet Nam  South-easternAsia   
193                               Yemen        WesternAsia   
194                              Zambia      EasternAfrica   
195                            Zimbabwe      EasternAfrica   

     Surface area (km2)  Population in thousands (2017)  \
0                652864                           35530   
1                 28748                            2930   
2               2381741                           41318   
3               1246700                           29784   
4                   442                             102   
..                  ...                             ...   
191              912050                           31977   
192              330967                           95541   
193              527968                           28250   
194              752612                           17094   
195              390757                           16530   

     GDP per capita (current US$)  Agricultural production index   \
0                           623.2                             125   
1                          3984.2                             134   
2                          4154.1                             160   
3                          4714.1                             175   
4                         14764.5                              88   
..                            ...                             ...   
191                       11068.9                             118   
192                        2067.9                             136   
193                        1106.4                             136   
194                        1311.1                             179   
195                         890.4                              99   

     International trade: Imports (million US$)  
0                                          3568  
1                                          4669  
2                                         47091  
3                                          8790  
4                                           491  
..                                          ...  
191                                       16324  
192                                      174111  
193                                        6861  
194                                        7442  
195                                        5212  

[196 rows x 7 columns]

# 1. Find out how many countries are there in the dataset.

In [3]:
data['country'].count()

196

# 2. Find out how many countries with respect to the region are available in the dataset

In [4]:
data.groupby('Region', as_index=False).country.count()

Region  country
0           Caribbean       16
1      CentralAmerica        8
2         CentralAsia        5
3       EasternAfrica       17
4         EasternAsia        7
5       EasternEurope       10
6           Melanesia        5
7          Micronesia        4
8        MiddleAfrica        9
9      NorthernAfrica        5
10    NorthernAmerica        4
11     NorthernEurope       10
12            Oceania        2
13          Polynesia        5
14  South-easternAsia       11
15       SouthAmerica       12
16     SouthernAfrica        5
17       SouthernAsia        9
18     SouthernEurope       12
19      WesternAfrica       15
20        WesternAsia       18
21      WesternEurope        7

# 3. On average which region has the highest

#### a. GDP per capita (current US$)

In [5]:
p=data.groupby('Region', as_index=False)[['GDP per capita (current US$)']].mean()
q=p.sort_values(by=['GDP per capita (current US$)'], ascending=False)
q.head(1)

Region  GDP per capita (current US$)
10  NorthernAmerica                       57659.0

#### On average, the region that has the highest GDP per capita, is NorthernAmerica (57659.0).

#### b. International trade: Imports (million US$)

In [6]:
r=data.groupby('Region', as_index=False)[['International trade: Imports (million US$)']].mean()
s=r.sort_values(by=['International trade: Imports (million US$)'], ascending=False)
s.head(1)

Region  International trade: Imports (million US$)
10  NorthernAmerica                                   663552.25

#### On average, the region that has the highest International trade: Imports, is NorthernAmerica (663552.25).

# 4. Find out which region has more consistent with respect to GDP per capita (current US$)

In [7]:
t=data.groupby('Region', as_index=False)[['GDP per capita (current US$)']].std()
u=t.sort_values(by=['GDP per capita (current US$)'])
u.head(1)

Region  GDP per capita (current US$)
19  WesternAfrica                    815.095399

#### The region which has more consistent GDP per capita, is WesternAfrica (815.095399).